apps/ex_nvr_web/lib/ex_nvr_web/controllers/api/user_controller.ex

```diff
@@ -0,0 +1,63 @@
+defmodule ExNVRWeb.API.UserController do
+  @moduledoc false
+
+  use ExNVRWeb, :controller
+
+  action_fallback ExNVRWeb.API.FallbackController
+
+  plug ExNVRWeb.Plug.User, [field_name: "id"] when action in [:update, :show, :delete]
+
+  import ExNVR.Authorization
+
+  alias ExNVR.Accounts
+  alias Plug.Conn
+
+  @spec create(Conn.t(), map()) :: Conn.t() | {:error, Ecto.Changeset.t()}
+  def create(%Conn{} = conn, params) do
+    user = conn.assigns.current_user
+
+    with :ok <- authorize(user, :user, :create),
```

Comment:
You can extract this instruction to a local plug too since it's used in all the endpoints.